src/refs.c

```diff
@@ -1130,6 +1130,18 @@ int git_reference__update_terminal(
 	return error;
 }
 
+const char *git_reference__commit_type(const git_commit *commit)
```

Comment:
Ahh, it will take me a little while longer to add some tests, I'll look into what's needed, thanks!